h/views/admin_groups.py

```diff
@@ -61,3 +68,38 @@ def on_success(appstruct):
 
     def _template_context(self):
         return {'form': self.form.render()}
+
+
+@view_defaults(route_name='admin_groups_edit',
+               permission='admin_groups')
+class GroupEditController(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @view_config(request_method='POST',
+                 request_param='delete')
+    def delete(self):
+        """Process submitted delete-group form"""
+        group = _form_request_group(self.request)
+
+        self.request.session.flash('Group "{name}" will be deleted'.format(name=group.name), 'success')
+        return HTTPFound(location=self.request.route_path('admin_groups'))
+
+
+@view_config(context=GroupNotFoundError)
```

Comment:
I copied this pattern from another view. Works great, but is pretty magic to me!